src/XComponent.Functions/Core/IFunctionsFactory.cs

```diff
@@ -7,5 +8,7 @@ public interface IFunctionsFactory {
         FunctionParameter GetTask(string componentName, string stateMachineName);
         void AddTaskResult(FunctionResult result);
         FunctionsConfiguration Configuration { get; set; }
+        void SaveRessources(string componentName, string key, string value);
+        List<FunctionsFactory.KeyValuePairSettingsItem> GetKeyValuePairs();
```

Comment:
`FunctionsFactory.KeyValuePairSettingsItem`

you add a dependency from the interface to the implementation, that's not good.
move `KeyValuePairSettingsItem` to an independent class